notebook/static/notebook/js/menubar.js

```diff
@@ -191,96 +187,56 @@ define([
-        });
-        this.element.find('#to_markdown').click(function () {
-            that.notebook.to_markdown();
-        });
-        this.element.find('#to_raw').click(function () {
-            that.notebook.to_raw();
-        });
-        
-        this.element.find('#toggle_current_output').click(function () {
-            that.notebook.toggle_output();
-        });
-        this.element.find('#toggle_current_output_scroll').click(function () {
-            that.notebook.toggle_output_scroll();
-        });
-        this.element.find('#clear_current_output').click(function () {
-            that.notebook.clear_output();
-        });
+             
+        var id_actions_array = [
+            ['#save_checkpoint', 'ipython.save-notebook'],
+            ['#restart_kernel', 'ipython.restart-kernel'],
+            ['#int_kernel', 'ipython.interrupt-kernel'],
+            ['#cut_cell', 'ipython.cut-selected-cell'],
+            ['#copy_cell', 'ipython.copy-selected-cell'],
+            ['#delete_cell', 'ipython.delete-cell'],
+            ['#undelete_cell', 'ipython.undo-last-cell-deletion'],
+            ['#split_cell', 'ipython.split-cell-at-cursor'],
+            ['#merge_cell_above', 'ipython.merge-selected-cell-with-cell-before'],
+            ['#merge_cell_below', 'ipython.merge-selected-cell-with-cell-after'],
+            ['#move_cell_up', 'ipython.move-selected-cell-up'],
+            ['#move_cell_down', 'ipython.move-selected-cell-down'],
+            ['#toggle_header', 'ipython.toggle-header'],
+            ['#toggle_toolbar', 'ipython.toggle-toolbar'],
+            ['#insert_cell_above', 'ipython.insert-cell-before'],
+            ['#insert_cell_below', 'ipython.insert-cell-after'],
+            ['#run_cell', 'ipython.execute-in-place'],
+            ['#run_cell_select_below', 'ipython.run-select-next'],
+            ['#run_cell_insert_below', 'ipython.execute-and-insert-after'],
+            ['#run_all_cells', 'ipython.run-all-cells'],
+            ['#run_all_cells_above', 'ipython.run-all-cells-above'],
+            ['#run_all_cells_below', 'ipython.run-all-cells-below'],
+            ['#to_code', 'ipython.change-selected-cell-to-code-cell'],
+            ['#to_markdown', 'ipython.change-selected-cell-to-markdown-cell'],
+            ['#to_raw', 'ipython.change-selected-cell-to-raw-cell'],
+            ['#toggle_current_output', 'ipython.toggle-output-visibility-selected-cell'],
+            ['#toggle_current_output_scroll', 'ipython.toggle-output-scrolling-selected-cell'],
+            ['#clear_current_output', 'ipython.clear-output-selected-cell'],
+        ];
+
+        for(var idx=0; idx< id_actions_array.length; idx++){
+            var id_act = id_actions_array[idx];
+            if(!that.actions.exists(id_act[1])){
+                console.warn('actions', id_act[1], 'does not exist, still binding it in case it will be defined later...' )
+            }
+            // IIFE cause F-you JS. 
```

Comment:
Haha, had to lookup IIFE acronym - also, probably don't want "F-you" in the comment, even though I totally agree, it's super annoying!